smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -0,0 +1,191 @@
+    private String paginationType;
+    private String interfaceLocation;
+
+    PaginationGenerator(Model model,
+                        ServiceShape service,
+                        OperationShape operation,
+                        SymbolProvider symbolProvider,
+                        TypeScriptWriter writer,
+                        String nonModularServiceName) {
+
+        this.symbolProvider = symbolProvider;
+        this.writer = writer;
+
+        this.serviceSymbol = symbolProvider.toSymbol(service);
+        this.operationSymbol = symbolProvider.toSymbol(operation);
+        this.inputSymbol = symbolProvider.toSymbol(operation).expectProperty("inputType", Symbol.class);
+        this.outputSymbol = symbolProvider.toSymbol(operation).expectProperty("outputType", Symbol.class);
+
+        String operationName = operation.getId().getName();
+        this.nonModularServiceName = nonModularServiceName;
+        this.methodName = Character.toLowerCase(operationName.charAt(0))
+                + operationName.substring(1); // e.g. listObjects
+        this.paginationType = this.nonModularServiceName + "PaginationConfiguration";
+        this.interfaceLocation = PaginationGenerator.generateInterfaceFilelocation();
+
+        PaginatedIndex paginatedModel = model.getKnowledge(PaginatedIndex.class);
+        Optional<PaginationInfo> paginationInfo = paginatedModel.getPaginationInfo(service, operation);
+        this.paginatedInfo = paginationInfo.orElseThrow(() -> {
+            return new CodegenException("Expected Paginator to have pagination information.");
+        });
+    }
+
+    @Override
+    public void run() {
+        // Import Service Types
+        writer.addImport(this.operationSymbol.getName(),
+                this.operationSymbol.getName(),
+                this.operationSymbol.getNamespace());
+        writer.addImport(this.inputSymbol.getName(),
+                this.inputSymbol.getName(),
+                this.inputSymbol.getNamespace());
+        writer.addImport(this.outputSymbol.getName(),
+                this.outputSymbol.getName(),
+                this.outputSymbol.getNamespace());
+        String nonModularLocation = this.serviceSymbol.getNamespace()
+                .replace(this.serviceSymbol.getName(), this.nonModularServiceName);
+        writer.addImport(this.nonModularServiceName,
+                this.nonModularServiceName,
+                nonModularLocation);
+        writer.addImport(this.serviceSymbol.getName(), this.serviceSymbol.getName(), this.serviceSymbol.getNamespace());
+
+        // Import Pagination types
+        writer.addImport("Paginator", "Paginator", "@aws-sdk/types");
+        writer.addImport(this.paginationType, this.paginationType, "./" + this.interfaceLocation.replace(".ts", ""));
+
+        this.writeClientSideRequest();
+        this.writeFullRequest();
+        this.writePager();
+    }
+
+    public static String generateOutputFilelocation(OperationShape operation) {
```

Comment:
These methods can be made package private by removing the `public` from their signature.

Reply:
nice catch. Done!